## **Analisis Exploratorio**
### **Desigualdad Educativa en Chile:** Análisis de Resultados PAES por Tipo de Establecimiento

#### **Nombres:** Lucas Chicao - Ninoska Cid - Isidora Gomez

#### **Curso**: IMT-2200 - Introduccion a la Ciencias de Datos

___
### **Contexto del proyecto:** 
La PAES es la prueba que miles de estudiantes chilenos rinden para acceder a la universidad, pero es un hecho que no todos cuentan con las mismas oportunidades educativas, ya que estas dependen del tipo de colegio en el que estudiaron.

En este proyecto, nos proponemos investigar si efectivamente hay diferencias en los puntajes de la PAES entre estudiantes de colegios públicos, subvencionados y privados, y también medir la magnitud de esas posibles diferencias.


### **Preguntas a responder:**
-¿Hay diferencias en los puntajes entre colegios públicos, subvencionados y privados?

-¿Qué tipo de colegio tiene mejores resultados en promedio?

-¿Qué tan grande es la brecha entre estos grupos?

-¿Existe correlación entre el tipo de establecimiento y las probabilidades de acceso a universidades tradicionales?

-¿Los resultados confirman las desigualdades educativas documentadas en otros estudios?

-...

### **Datos a usar:**
Los datos que usaremos en nuestro proyecto seran principalmente de base de datos de:

-DEMRE

-chiquillos si les parece aqui hay una base de datos para poder usar, si lo usamos eliminen este mensajito: https://www.kaggle.com/datasets/daniellopez01/admisionuescl

### **Analisis Exploratorio:**
En las siguientes lineas exploraremos los datos que ya obtuvimos y revisaremos como vamos en relacion con datos.
 

In [2]:
#LIBRERIAS A USAR
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import os
#...

# Analisis Exploratorio de la base de datos del DEMRE (2023 - 2025)

La bdd del demre viene con 4 tipos de archivos segun el año

    - Inscripcion   

    - Matricula

    - Postulacion

    - Rendicion

Cada uno de estos tipo de archivos tienen distinto formato

In [29]:
#CARGAR DATOS Y DF (INSCRIPCION)



base_path = os.path.dirname(os.getcwd())  # Esto sube un nivel desde 'Análisis Exploratorio'

file_path = os.path.join(base_path, 'DEMRE_bdd', '2023','Incripcion','Inscritos_Admisión2023','ArchivoB_Adm2023.csv')

df_2023_inscripcion = pd.read_csv(file_path, sep=';')
df_2023_inscripcion


ID_aux  ANYO_PROCESO  SEXO  RBD  COD_ENS  REGIMEN  \
0       id_9265592376510          2023     1  1.0    510.0      3.0   
1       id_9492594679209          2023     1  1.0    510.0      3.0   
2       id_9325593372588          2023     1  1.0    510.0      3.0   
3       id_3443934474359          2023     1  1.0    510.0      3.0   
4       id_9355593175583          2023     1  1.0    510.0      3.0   
...                  ...           ...   ...  ...      ...      ...   
293860  id_9162591776233          2023     1  NaN      NaN      NaN   
293861  id_6620966272093          2023     2  NaN      NaN      NaN   
293862  id_8886588078640          2023     2  NaN      NaN      NaN   
293863  id_3253532675336          2023     2  NaN      NaN      NaN   
293864  id_8843588774312          2023     1  NaN      NaN      NaN   

       RAMA_EDUCACIONAL  GRUPO_DEPENDENCIA  FECHA_NACIMIENTO  ANYO_EGRESO  \
0                    T2                4.0             92004       2022.0   
1                    T2                4.0             62004       2021.0   
2                    T2                4.0             22005       2022.0   
3                    T2                4.0             61992       2010.0   
4                    T2                4.0            122004       2022.0   
...                 ...                ...               ...          ...   
293860              NaN                NaN            112003       2022.0   
293861              NaN                NaN             61998       2016.0   
293862              NaN                NaN             61999       2019.0   
293863              NaN                NaN            102002       2021.0   
293864              NaN                NaN            101998       2019.0   

        ...  CODIGO_COMUNA  CODIGO_REGION_D  CODIGO_COMUNA_D  \
0       ...        15101.0               15            15101   
1       ...        15101.0               15            15101   
2       ...        15101.0               15            15101   
3       ...        15101.0               15            15101   
4       ...        15101.0               15            15101   
...     ...            ...              ...              ...   
293860  ...            NaN               13            13125   
293861  ...            NaN                4             4101   
293862  ...            NaN                9             9101   
293863  ...            NaN                1             1101   
293864  ...            NaN                8             8105   

        SITUACION_EGRESO  RINDIO_PROCESO_ANTERIOR  RINDIO_PROCESO_ACTUAL  BEA  \
0                      1                        0                      1  NaN   
1                      5                        1                      1  NaN   
2                      1                        0                      1  BEA   
3                      5                        0                      1  NaN   
4                      1                        0                      1  NaN   
...                  ...                      ...                    ...  ...   
293860                 2                        0                      1  NaN   
293861                 6                        0                      1  NaN   
293862                 6                        1                      3  NaN   
293863                 8                        0                      3  NaN   
293864                 6                        0                      2  NaN   

        PACE PAIS_NACIMIENTO INGRESO_PERCAPITA_GRUPO_FA  
0       PACE             210                          2  
1        NaN             210                          2  
2       PACE             210                          2  
3        NaN             210                          1  
4       PACE             210                         99  
...      ...             ...                        ...  
293860   NaN             210                         99  
293861   NaN             210      

In [44]:
#EXPLORACION 
df_2023_inscripcion.describe(include='all') 

ID_aux  ANYO_PROCESO           SEXO            RBD  \
count             293865      293865.0  293865.000000  291232.000000   
unique            293865           NaN            NaN            NaN   
top     id_9265592376510           NaN            NaN            NaN   
freq                   1           NaN            NaN            NaN   
mean                 NaN        2023.0       1.539775   11984.058747   
std                  NaN           0.0       0.498416    9151.345931   
min                  NaN        2023.0       1.000000       1.000000   
25%                  NaN        2023.0       1.000000    4667.000000   
50%                  NaN        2023.0       2.000000    9693.000000   
75%                  NaN        2023.0       2.000000   17736.000000   
max                  NaN        2023.0       2.000000   99999.000000   

              COD_ENS        REGIMEN RAMA_EDUCACIONAL  GRUPO_DEPENDENCIA  \
count   291232.000000  269287.000000           291232      291232.000000   
unique            NaN            NaN                7                NaN   
top               NaN            NaN               H1                NaN   
freq              NaN            NaN           199821                NaN   
mean       365.715515       2.866184              NaN           2.283801   
std        100.190229       0.439937              NaN           0.704313   
min        310.000000       1.000000              NaN           1.000000   
25%        310.000000       3.000000              NaN           2.000000   
50%        310.000000       3.000000              NaN           2.000000   
75%        410.000000       3.000000              NaN           3.000000   
max        910.000000       3.000000              NaN           4.000000   

        FECHA_NACIMIENTO    ANYO_EGRESO  ...  CODIGO_COMUNA  CODIGO_REGION_D  \
count      293865.000000  293531.000000  ...  291232.000000    293865.000000   
unique               NaN            NaN  ...            NaN              NaN   
top                  NaN            NaN  ...            NaN              NaN   
freq                 NaN            NaN  ...            NaN              NaN   
mean        67379.235751    2021.114407  ...    9741.656514         9.555701   
std         34725.991403       2.441968  ...    3917.504439         3.932259   
min         11955.000000    1979.000000  ...    1101.000000         1.000000   
25%         32005.000000    2021.000000  ...    6117.000000         6.000000   
50%         72000.000000    2022.000000  ...   10101.000000        10.000000   
75%        101998.000000    2022.000000  ...   13122.000000        13.000000   
max        122006.000000    2022.000000  ...   16305.000000        16.000000   

        CODIGO_COMUNA_D  SITUACION_EGRESO  RINDIO_PROCESO_ANTERIOR  \
count     293865.000000     293865.000000            293865.000000   
unique              NaN               NaN                      NaN   
top                 NaN               NaN                      NaN   
freq                NaN               NaN                      NaN   
mean        9754.224300          2.327865                 0.208899   
std         3913.177912          1.891272                 0.406523   
min         1101.000000          1.000000                 0.000000   
25%         6116.000000          1.000000                 0.000000   
50%        10102.000000          1.000000                 0.000000   
75%        13123.000000          5.000000                 0.000000   
max        16305.000000          8.000000                 1.000000   

        RINDIO_PROCESO_ACTUAL    BEA   PACE PAIS_NACIMIENTO  \
count           293865.000000  14403  37338   293865.000000   
unique                    NaN      1      1             NaN   
top                       NaN    BEA   PACE             NaN   
freq                      NaN  14403  37338             NaN   
mean                 1.151889    NaN    NaN      210.032484   
std                  0.592508    NaN    NaN       13.449517 

In [ ]:
'''
count	Conteo	Número de valores no nulos en la columna
unique	Únicos	Número de valores distintos (solo para columnas tipo objeto/categoría)
top	Más común	Valor que aparece con mayor frecuencia (modo)
freq	Frecuencia	Cuántas veces aparece el valor más común (top)
mean	Media	Promedio aritmético de los valores numéricos
std	Desviación estándar	Medida de dispersión: qué tan lejos están los datos del promedio
min	Mínimo	Valor más bajo en la columna
25%	Cuartil 1	Valor por debajo del cual está el 25% de los datos
50%	Mediana	Valor central: 50% de los datos están por debajo
75%	Cuartil 3	Valor por debajo del cual está el 75% de los datos
max	Máximo	Valor más alto en la columna
'''

'\ncount\tConteo\tNúmero de valores no nulos en la columna\n\nunique\tÚnicos\tNúmero de valores distintos (solo para columnas tipo objeto/categoría)\n\ntop\tMás común\tValor que aparece con mayor frecuencia (modo)\n|\nfreq\tFrecuencia\tCuántas veces aparece el valor más común (top)\n\nmean\tMedia\tPromedio aritmético de los valores numéricos\n\nstd\tDesviación estándar\tMedida de dispersión: qué tan lejos están los datos del promedio\n\nmin\tMínimo\tValor más bajo en la columna\n\n25%\tCuartil 1\tValor por debajo del cual está el 25% de los datos\n\n50%\tMediana\tValor central: 50% de los datos están por debajo\n\n75%\tCuartil 3\tValor por debajo del cual está el 75% de los datos\n\nmax\tMáximo\tValor más alto en la columna\n'


---------------------------------------------

# SEXO: codificado como 1 (Hombre), 2 (Mujer)

Media: 1.54 → indica ligera mayoría femenina.

Mediana: 2.0 → la mitad son mujeres o más.

Distribución esperada: ~46% hombres, ~54% mujeres.

---------------------------------------------
# Establecimiento de egreso
RBD: código del establecimiento (media ≈ 11.984, std ≈ 9.151) → gran dispersión, muchos establecimientos distintos.

COD_ENS: código de enseñanza MINEDUC (media ≈ 365.7) → predominan códigos bajos (310 es el mínimo).

REGIMEN: 1 = Masculino, 2 = Femenino, 3 = Coeducacional

Media ≈ 2.86 → predominan los coeducacionales.

Mediana = 3 → más del 50% egresan de régimen mixto.

RAMA_EDUCACIONAL: 7 valores únicos, top = H1 → mayoría egresa de enseñanza humanista científico diurna

---------------------------------------------
# Dependencia administrativa
GRUPO_DEPENDENCIA:

Media ≈ 2.28 → predominan establecimientos subvencionados.

Rango: 1 (particular pagado) a 4 (Servicio Local de Educación)

---------------------------------------------

# Edad y egreso
FECHA_NACIMIENTO: media ≈ 67.379 → codificación tipo MMYYYY, por ejemplo 072000 (julio 2000)

ANYO_EGRESO: media ≈ 2021.1, min = 1979 → hay postulantes egresados hace décadas.

Mediana = 2022 → mayoría egresó recientemente.

---------------------------------------------
# Ubicación
CODIGO_REGION_D y CODIGO_COMUNA_D: códigos de domicilio

Media región ≈ 9.55 → predominan regiones centro-sur.

Comuna media ≈ 9754 → gran dispersión territorial.

---------------------------------------------
# Situación académica
SITUACION_EGRESO: codificada del 1 al 8

Media ≈ 2.33 → predominan casos de promoción nacional regular o validación.

Mediana = 1 → mayoría son alumnos regulares del año.

---------------------------------------------
# Participación en procesos
RINDIO_PROCESO_ANTERIOR:

Media ≈ 0.21 → pocos rindieron pruebas en 2022.

79% no participó en el proceso anterior.

RINDIO_PROCESO_ACTUAL:

Media ≈ 1.15 → mayoría inscribió al menos una prueba.

Rango: 0 (no inscribió) a 3 (ambas pruebas)

---------------------------------------------
# Programas especiales
BEA y PACE:

BEA: 14.403 postulantes (~4.9%) preseleccionados para Beca Excelencia Académica.

PACE: 37.338 (~12.7%) pertenecen a establecimientos PACE.

---------------------------------------------
# Ingreso per cápita
INGRESO_PERCAPITA_GRUPO_FA:

Codificado del 1 al 10 (decil), más 99 = "Prefiero no responder"

Media ≈ 27.59 → indica presencia significativa del valor 99.

Mediana = 5 → distribución centrada en deciles medios.

Desviación alta (≈ 40.93) → mezcla entre deciles bajos y respuestas omitidas.

---------------------------------------------
# Resumen 
El dataframe

In [45]:
#EXPLORACION 
print(f'Columnas : {df_2023_inscripcion.columns}')
print('-----------')
print(f'Filas : {df_2023_inscripcion.shape[0]}, Columnas : {df_2023_inscripcion.shape[1]}') #Filas, Columnas
print('-----------')


Columnas : Index(['ID_aux', 'ANYO_PROCESO', 'SEXO', 'RBD', 'COD_ENS', 'REGIMEN',
       'RAMA_EDUCACIONAL', 'GRUPO_DEPENDENCIA', 'FECHA_NACIMIENTO',
       'ANYO_EGRESO', 'CODIGO_REGION', 'CODIGO_PROVINCIA', 'CODIGO_COMUNA',
       'CODIGO_REGION_D', 'CODIGO_COMUNA_D', 'SITUACION_EGRESO',
       'RINDIO_PROCESO_ANTERIOR', 'RINDIO_PROCESO_ACTUAL', 'BEA', 'PACE',
       'PAIS_NACIMIENTO', 'INGRESO_PERCAPITA_GRUPO_FA'],
      dtype='object')
-----------
Filas : 293865, Columnas : 22
-----------


------------------------------------------------------------------------------------------------------------------------

In [ ]:
#CARGAR DATOS Y DF


In [ ]:
#EXPLORACION 


In [ ]:
#LIMPIEZA DE DATOS


In [ ]:
#ESTADISTICAS DE DATOS (tipo promedios colegios, comunas con mas puntos, ...)


In [ ]:
#VISUALIZACION DE LOS DATOS


### **Conclusion:** 
En relacion con los datos que analizamos anteriormente podemos darnos cuenta que...